drivers/media/platform/bcm2835/bcm2835-unicam.c

```diff
@@ -1996,13 +2268,20 @@ static int register_node(struct unicam_device *unicam, struct unicam_node *node,
 	spin_lock_init(&node->dma_queue_lock);
 	mutex_init(&node->lock);
 
-	if (type == V4L2_BUF_TYPE_VIDEO_CAPTURE) {
+	if (pad_id == IMAGE_PAD) {
 		/* Add controls from the subdevice */
 		ret = v4l2_ctrl_add_handler(&node->ctrl_handler,
 					    unicam->sensor->ctrl_handler, NULL,
 					    true);
 		if (ret < 0)
 			return ret;
+
+		/*
+		 * If the source has no controls then remove our
+		 * ctrl handler.
+		 */
+		if (list_empty(&node->ctrl_handler.ctrls))
+			unicam->v4l2_dev.ctrl_handler = NULL;
```

Comment:
Is this the right thing to be NULLing? I didn't think it was set, and you've been setting it fh_open.

Reply:
Actually I think you want
```
		if (!list_empty(&node->ctrl_handler.ctrls))
			vdev->ctrl_handler = &node->ctrl_handler;
```
having first moved the assignment of vdev from a dozen lines later to just before this.

Reply:
Done.